### PR TITLE
Feature / RadioRichButtons

### PR DESCRIPTION
**Pull request by enguerranws**
Todo:
- [x] add the story

### CODE REVIEW

File: src/shared/Fieldset.tsx [hunk 1]
```diff
@@ -14,27 +13,21 @@ import { cx } from "../tools/cx";
...
         className?: string;
         id?: string;
         classes?: Partial<Record<"root" | "legend" | "content", string>>;
         style?: CSSProperties;
         legend?: ReactNode;
         hintText?: ReactNode;
+        options: {
+            label: ReactNode;
+            hintText?: ReactNode;
+            nativeInputProps: ComponentProps<"input">;
```
Comment (garronej) [inline src/shared/Fieldset.tsx:29]: Here it's best to use the `ComponetProps` type helper from React.

**Comment by garronej** [comment]
Hey @enguerranws,  

Thanks a lot for the PR!   

I first I did this review to improve consistency with the rest of the project: https://github.com/codegouvfr/react-dsfr/pull/211/commits/765f23343f8109d18a094819f1dbc67f96252a66  

But in the end I did even simpler.  

I don't think we need to have a separate RadiRichButton.  We can just infer it's rich by the use of illustations.  
The drawback is that we can't enforce at the type level that if there is one radio that has an illustration all radio must have one but I think it's really no big deal.

**Comment by enguerranws** [comment]
> I don't think we need to have a separate RadiRichButton. We can just infer it's rich by the use of illustations.
> The drawback is that we can't enforce at the type level that if there is one radio that has an illustration all radio must have one but I think it's really no big deal.

I thought about that, but I wanted to stick the DSFR's components (where Radio buttons and Radio rich are presented as different components), but that's fine for me!

**Comment by garronej** [comment]
> I thought about that, but I wanted to stick the DSFR's components (where Radio buttons and Radio rich are presented as different components), but that's fine for me!  

Oh! Is that so?  In that case you probably made the right call.  
Anyway, now that it's published do you think I should rollback or keep it like this?

**Comment by garronej** [comment]
I can rollback the last commit if you think it's best. Idk

**Pull request merged**